Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitClient.swift

```diff
@@ -59,7 +58,7 @@ import BraintreeCore
                 return
             } else if let url = URL(string: createMandateResult.approvalURL) {
                 self.apiClient.sendAnalyticsEvent("ios.sepa-direct-debit.create-mandate.success")
-                self.startAuthenticationSession(url: url, context: context) { success, error in
+                self.startAuthenticationSession(url: url, context: self) { success, error in
```

Comment:
Might this cause a retain cycle? Between the SEPAClient.swift & the ASWebAuthSession?

We have[ this note](https://github.com/braintree/braintree_ios/blob/2a0a3ce2143bf8c3ce9194a9f06fb794f631f29f/Sources/BraintreePayPal/BTPayPalClient.m#L231-L232) for PayPalClient, which originated in this PR https://github.com/braintree/braintree_ios/pull/596.

Reply:
Good catch - yeah we can add this. It looks like the `authenticationSession` property is coupled to the `isAuthenticationSessionStarted` property in `BTPayPalClient`. Is that something we want to add as well to SEPA (looks like it is used to track if the auth session was started successfully since it would need to be optional to set to `nil` and prevent the retain cycle). In SEPA this is currently not optional so would need to make some updates, so may be good to also track this if we wanted.

Reply:
So going through the SEPA and PayPal flows a few times and using the Xcode memory graph debugger, I'm not able to replicate the same leak reported in that original GH Issue, using the code as-is on this branch.

I am not sure exactly why this is - but a handful has changed since that original bug. Both the demo & module are in Swift now + we have the wrapper WebAuthenticationSession around Apple's ASWeb, which could have something to do with it.

Reply:
Regarding `isAuthenticationSessionStarted`, I was just curious and ran a quick SPLUNK. `1.7%` of these ASWeb events are failures over the last day. I guess this isn't far off of what I expected, but I expected it to be slightly lower.

<img width="1711" alt="Screen Shot 2023-01-19 at 11 55 06 AM" src="https://user-images.githubusercontent.com/35243507/213522672-80f5c8dd-f0c1-4385-8bad-c371f91a7b43.png">

In terms of adding these kinds of analytics for SEPA, on one hand, it is nice to maintain parity with our other ASWeb flows (ex: PayPal), but I know we have a history of tracking too much. I'd vote to add it in for now to maintain parity. Maybe at some point we'll audit the whole SDK analytics to streamline ... but I don't feel too strongly. Thoughts?

Reply:
Interesting that you aren't seeing it in the memory graphs. I tried poking around online to see if there were some updates somewhere but am coming up empty. But yeah could have also been tied to something else. 🤷 

I think that makes sense, we can add for now, make it `nil` just incase there is something that could go sideways somewhere. And yeah, I think for now we add the events so we have parity and we can always get rid of it down the road... or if it makes our life a pain with FPTI. 😅

Reply:
What does the leak look like in the memory graph debugger? I don't think I see it on `5.x` either.

Reply:
> What does the leak look like in the memory graph debugger? I don't think I see it on 5.0.0-beta2 either.

If I comment out [this line](https://github.com/braintree/braintree_ios/blob/2a0a3ce2143bf8c3ce9194a9f06fb794f631f29f/Sources/BraintreePayPal/BTPayPalClient.m#L231-L232) (basically revert [this PR](https://github.com/braintree/braintree_ios/pull/596)), you can trigger the leak.

It will look like this after 1 PayPal checkout:
<img width="276" alt="Screen Shot 2023-01-19 at 9 43 02 PM" src="https://user-images.githubusercontent.com/35243507/213613285-d84ce55e-b565-4f98-beac-2ba4f505e666.png">

And then after going through the PayPal flow 5 times it will look like this:
<img width="281" alt="Screen Shot 2023-01-19 at 9 42 40 PM" src="https://user-images.githubusercontent.com/35243507/213613371-613fc8ee-cb83-481a-a925-a74dd946421d.png">
With the fix, the count of `BTPayPalClient` and `BTPayPalCheckoutRequest` will _not_ continue increasing, and will just remain at 1.

Reply:
Since I can't trigger any issues with the current implementation, I don't think we need to make changes to the PR. I just don't really know exactly _why_ it is no longer an issue in this Swift version.

Also happy to go through it together, might be easier to make sure we're not missing something.